File: ApplozicCocoaPodDemo/ApplozicCocoaPodDemo/ViewController.m
//
//  ViewController.m
//  ApplozicCocoaPodDemo
//
//  Created by Abhishek Thapliyal on 9/8/16.
//  Copyright © 2016 Abhishek Thapliyal. All rights reserved.
//

#import "ViewController.h"
#import "ALChatManager.h"
#import <Applozic/Applozic.h>
#import "ChatOptionVC.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UITextField *userIdTxtField;
@property (weak, nonatomic) IBOutlet UITextField *emailTxtField;
@property (weak, nonatomic) IBOutlet UITextField *paswordField;
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicator;

- (IBAction)loginAction:(id)sender;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (IBAction)loginAction:(id)sender
{
    
    if(!self.userIdTxtField.text.length || !self.paswordField.text.length)
    {
        [ALUtilityClass showAlertMessage:@"Invalid userId/Password" andTitle:@"Oops!!!"];
        return;
    }
    
    [self.activityIndicator startAnimating];
    ALUser *user = [ALUser new];
    
    user.userId = self.userIdTxtField.text;
    user.password = self.paswordField.text;
    user.email = self.emailTxtField.text;
    
    [ALUserDefaultsHandler setUserId:user.userId];
    [ALUserDefaultsHandler setPassword:user.password];
    [ALUserDefaultsHandler setEmailId:user.email];
    
    ALChatManager *manager = [ALChatManager new];
    [manager registerUserWithCompletion:user withHandler:^(ALRegistrationResponse *rResponse, NSError *error) {
        
        UIStoryboard *storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
        ChatOptionVC * chatOptionVC = [storyBoard instantiateViewControllerWithIdentifier:@"ChatOptionVC"];
        
        chatOptionVC.alUser = user;
        [self.activityIndicator stopAnimating];
        [self presentViewController:chatOptionVC animated:YES completion:nil];
    }];

}


@end

File: sampleapp-completeswift/sampleapp-completeswift/ViewController.swift
//
//  ViewController.swift
//  sampleapp-completeswift
//
//  Created by Mukesh Thawani on 04/05/17.
//  Copyright © 2017 Applozic. All rights reserved.
//

import UIKit
import Applozic

class ViewController: UIViewController {

    override func viewDidAppear(_ animated: Bool) {
//        registerAndLaunch()

    }

    override func viewDidLoad() {
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    @IBAction func logoutAction(_ sender: UIButton) {
        let registerUserClientService: ALRegisterUserClientService = ALRegisterUserClientService()
        registerUserClientService.logout { (response, error) in

        }
        self.dismiss(animated: false, completion: nil)
    }

    @IBAction func launchChatList(_ sender: Any) {
        let conversationVC = ConversationListViewController()
        let nav = UINavigationController(rootViewController: conversationVC)
        self.present(nav, animated: false, completion: nil)
    }
}

